Segment-ComScore/Classes/SEGComScoreIntegration.m

```diff
@@ -15,45 +15,47 @@ - (instancetype)initWithSettings:(NSDictionary *)settings
 {
     if (self = [super init]) {
         self.settings = settings;
-        self.comScoreClass = [CSComScore class];
         
-        [self.comScoreClass setAppContext];
-        SEGLog(@"[CSComScore setAppContext]");
-        [self.comScoreClass setCustomerC2:[self customerC2]];
-        SEGLog(@"[CSComScore setCustomerC2: %@]", [self customerC2]);
-        [self.comScoreClass setPublisherSecret:[self publisherSecret]];
-        SEGLog(@"[CSComScore setPublisherSecret: %@]", [self publisherSecret]);
-        [self.comScoreClass setAppName: [self appName]];
-        SEGLog(@"[CSComScore setAppName: %@]", [self appName]);
-        [self.comScoreClass setSecure: [self useHTTPS]];
-        SEGLog(@"[CSComScore setSecure: %@]", [self useHTTPS] ? @YES : @NO);
-        if ([self autoUpdate]) {
-            if ([self foregroundOnly]) {
-                [self.comScoreClass enableAutoUpdate:[self autoUpdateInterval] foregroundOnly: YES];
-                SEGLog(@"[CSComScore enableAutoUpdate: %@ foregroundOnly: YES]", [self autoUpdateInterval]);
+        SCORPublisherConfiguration *config = [SCORPublisherConfiguration publisherConfigurationWithBuilderBlock:^(SCORPublisherConfigurationBuilder * builder) {
+            builder.publisherId = settings[@"appName"]; // @ladan please config this is actually the case. Or we need new config with publisherId
+            builder.publisherSecret = settings[@"publisherSecret"];
+            
+            
+            if ([settings[@"autoUpdate"] boolValue] && [settings[@"foregroundOnly"] boolValue]) {
```

Comment:
Correct, it was just about being explicit. `NSNumber` is required to call `boolValue`. `[settings[@"autoUpdate"] boolValue]` is implicit in that you expect `settings[@"autoUpdate"]` to return an `NSNumber`.  In general I recommend casting because there often gotchas with implicit conversions.